src/nvim/os/env.c

```diff
@@ -456,6 +456,15 @@ char *vim_getenv(const char *name)
     if (p_hf != NULL && vim_strchr(p_hf, '$') == NULL) {
       vim_path = (char *)p_hf;
     }
+
+#ifdef WIN32
+    char exe_name[MAXPATHL];
+    if (vim_path == NULL) {
+      GetModuleFileName(NULL, exe_name, MAXPATHL);
+      vim_path = exe_name;
```

Comment:
`GetModuleFileName` is simply a `#define` for `GetModuleFileNameW` when `UNICODE` is defined when building on Windows. It's `GetModuleFileNameA` otherwise.

As for adding a `os_exepath` that is a wrapper for `uv_exepath` do we still want to only call that inside the `#ifdef WIN32` block here?